src/model/neural_network/neural_network.py

```diff
@@ -0,0 +1,104 @@
+from typing import List
+
+import numpy as np
+
+from model.neural_network.activation_function_utils import (
+    activation_functions_utils,
+    ActivationFunctionUtils,
+)
+
+np.set_printoptions(suppress=True)
+
+
+class NeuralNetworkHiddenLayerInfo:
+    def __init__(self, activation_function_name: str, neurons_count: int):
+        self.activation_function_name: str = activation_function_name
+        self.neurons_count: int = neurons_count
```

Comment:
This can be converted to a dataclass for simplicity
```suggestion
@dataclass
class NeuralNetworkHiddenLayerInfo:
    self.activation_function_name: str
    self.neurons_count: int
```